app/src/main/java/com/example/mac/chartr/activities/RegisterActivity.java

```diff
@@ -151,48 +161,121 @@ private void initSignup() {
                 email.setBackground(getDrawable(R.drawable.text_border_error));
+                Toast.makeText(getApplicationContext(), "Invalid Email", Toast.LENGTH_LONG)
+                .show();
                 return;
             }
             user.setEmail(emailInput);
 
+            /*
+             Check for a valid PASSWORD
+             */
             String userpasswordInput = password.getText().toString();
+            TextView passwordMessage = (TextView) findViewById(R.id.textViewUserRegPasswordMessage);
             userPasswd = userpasswordInput;
-            if (userpasswordInput.isEmpty()) {
-                TextView view = (TextView) findViewById(R.id.textViewUserRegPasswordMessage);
-                view.setText(password.getHint() + " cannot be empty");
+            if (userpasswordInput.isEmpty() || !passwordMessage.getText().toString().equals("")) {
                 password.setBackground(getDrawable(R.drawable.text_border_error));
+                Toast.makeText(getApplicationContext(), "Invalid Password", Toast.LENGTH_LONG)
+                        .show();
+                return;
+            }
+
+            /*
+             Check for a valid NAME
+             */
+            String name = givenName.getText().toString();
+            TextView nameMessage = (TextView) findViewById(R.id.textViewRegGivenNameMessage);
+            if (name.isEmpty() || !nameMessage.getText().toString().equals("")) {
+                givenName.setBackground(getDrawable(R.drawable.text_border_error));
+                Toast.makeText(getApplicationContext(), "Invalid Name", Toast.LENGTH_LONG)
+                        .show();
+                return;
+            }
+
+            /*
+             Check for a valid BIRTHDAY
+             */
+            String birthdayInput = givenName.getText().toString();
+            TextView birthdayMessage = (TextView) findViewById(R.id.textViewRegBirthdayMessage);
+            if (birthdayInput.isEmpty() || !birthdayMessage.getText().toString().equals("")) {
+                birthday.setBackground(getDrawable(R.drawable.text_border_error));
+                Toast.makeText(getApplicationContext(), "Invalid Birthday", Toast.LENGTH_LONG)
+                        .show();
+                return;
+            }
+
+            /*
+             Check for a valid PHONE NUMBER
+             */
+            String phoneInput = givenName.getText().toString();
+            TextView phoneMessage = (TextView) findViewById(R.id.textViewRegPhoneMessage);
+            if (phoneInput.isEmpty() || !phoneMessage.getText().toString().equals("")) {
+                phone.setBackground(getDrawable(R.drawable.text_border_error));
+                Toast.makeText(getApplicationContext(), "Invalid Phone Number",
+                        Toast.LENGTH_LONG)
+                        .show();
                 return;
             }
 
+            /*
+             Check name and add to AppHelper Attribute
+             */
```

Comment:
Aren't the following checks redundant after you put in the new ones?

Reply:
Yeah, we actually should be able to remove the "if", just keep the logic because if it makes it that far, the strings are going to be non-empty